implement setup0 type in code base, updates makefile deps.

garrettwrong opened the pull request:
I still need to review this, but is a good time to discuss.

Some of it is bloated, but I expect will collapse once all NML groups are done similarly.  Also a significant amount of common block would actually be replaced by scoped module(s)...

garrettwrong commented:
@fmpoli68 , please stop by when you have time to discuss. Thanks

garrettwrong commented on achief1.f90:
```diff
@@ -109,20 +110,20 @@ subroutine achief1
+    if (setup0%nmlstout.eq."enabled") then
        write(6,*)'  In achief1: '
-       write(6,setup0)
+       !name now private, can write from setter write(6,setup0)
```
figure out where we want the ability to dump this.

garrettwrong replied:
implimented print_setup0 in cql_conf module

garrettwrong commented on achief1.f90:
```diff
@@ -232,7 +233,7 @@ subroutine achief1
        n=n+1
        n_(1)=n ! new time-step for this flux surface
-       ! for 2-d (v_par,v_perp) calculation ntloop controls
+       ! for 2-d (v_par,v_perp) calculation ntloop controsetup0%ls
```
(gbw) fix this

garrettwrong commented on achiefn.f90:
```diff
@@ -89,7 +90,7 @@ subroutine achiefn(kopt)
            n=n+1
            n_(l_)=n ! new time-step for this flux surface
-           ! for 2-d (v_par,v_perp) calculation ntloop controls
+           ! for 2-d (v_par,v_perp) calculation ntloop controsetup0%ls
```
(gbw) fix this

garrettwrong commented on aindfpa.f90:
```diff
@@ -28,43 +29,43 @@ subroutine aindfpa
-      ioutput(1)=6
-      ioutput(2)=0
-      mnemonic="mnemonic"
+      setup0%ibox(1)="unset"
```
gbw, i think we can/should move these to the conf mod. If we get all of them I can ditch the local pointer hack...

garrettwrong commented on aingeom.f90:
```diff
@@ -27,11 +27,11 @@ subroutine aingeom
 !
-!     aingeom controls the flux surface geometry and sets up magnetic
-!     fields, bpsi=(B(z))/B(0)); it also calls the "eq"uilibrium module
+!     aingeom controsetup0%ls the flux surface geometry and sets up magnetic
```
(gbw) fix this

garrettwrong commented on aingeom.f90:
```diff
@@ -75,7 +75,7 @@ subroutine aingeom
            ! up-dn symmetry: everything is evaluated over 1/2 surface
            symm=2.d0
-           ! Factor symm=2 is because solrz(l,ll),solzz(l,ll) is
+           ! Factor symm=2 is because sosetup0%lrz(l,ll),solzz(l,ll) is
```
and this

garrettwrong commented on ainpltpa.f90:
```diff
@@ -180,7 +180,7 @@ subroutine ainpltpa
 
       write(t_,13027)
-13027 format("ndtr1a is maximum number of time step intervals dtr1().")
+13027 format("ndtr1a is maximum number of time step intervasetup0%ls dtr1().")
```
and this

garrettwrong commented on ainsetpa.f90:
```diff
@@ -19,18 +19,18 @@ subroutine ainsetpa
-!     (indxls labels the orbit position from 1 to ls).
-!     lrindx(1)=namelist input
-!     lrindx(2:ls)=lrindx(1), indxlr(lrindx(1))=1
+!     (indxlr labesetup0%ls the radial surfaces from 1 to setup0%lrz)
```
these

garrettwrong commented on ainsetva.f90:
```diff
@@ -141,24 +141,24 @@ subroutine ainsetva
          WRITE(*,225)
 !MPIINSERT_ENDIF_RANK
- 225     format(//,'ERROR:  Presently ndeltarho calls are only setup',/ &
+ 225     format(//,'ERROR:  Presently ndeltarho calsetup0%ls are only setup',/ &
```
here

garrettwrong commented on ainsetva.f90:
```diff
@@ -141,24 +141,24 @@ subroutine ainsetva
 !MPIINSERT_IF_RANK_EQ_0
          WRITE(*,227)
- 227     format(//,'ERROR:  Presently ndeltarho calls are only setup',/ &
+ 227     format(//,'ERROR:  Presently ndeltarho calsetup0%ls are only setup',/ &
```
again

garrettwrong commented on ampfar.f90:
```diff
@@ -51,7 +51,7 @@ real(c_double) function ampfarl(distn,ll)
       ampfarl= (4.d0*pi*(-charge*vnorm))/ &
 !BH131110     +        vnorm*drpmconz(ll)*dlpsii(ll)*cn
-!BH131110, BUT think vnorm* is correct, cancels vnorm in impanvc0:dfdvz &
+!BH131110, BUT think vnorm* is correct, cancesetup0%ls vnorm in impanvc0:dfdvz &
```
here

garrettwrong commented on cfpcoefn.f90:
```diff
@@ -140,7 +140,7 @@ subroutine cfpcoefn
         tam9(1)=0.
 !..................................................................
-!     tam2(jx) and tam6(jx) represent integrals from xmax to infinity
+!     tam2(jx) and tam6(jx) represent integrasetup0%ls from xmax to infinity
```
here

garrettwrong commented on cfpcoefn.f90:
```diff
@@ -428,13 +428,13 @@ subroutine cfpcoefn
 !     The do loops 302 through 307 seek to take advantage of the
 !     Maxwellian nature of f for v < vth/2. This is employed in
-!     the integrations to obtain the functionals - Thus f is
+!     the integrations to obtain the functionasetup0%ls - Thus f is
```
again

garrettwrong commented on cfpcoefr.f90:
```diff
@@ -190,7 +190,7 @@ subroutine cfpcoefr
         tam11(jx)=0.
 !......................................................................:
-!     tam3(jx)->tam6(jx), tam11(jx) and tam12(jx) represent integrals
+!     tam3(jx)->tam6(jx), tam11(jx) and tam12(jx) represent integrasetup0%ls
```
again

garrettwrong commented on cfpcoefr.f90:
```diff
@@ -424,7 +424,7 @@ subroutine cfpcoefr
 
 !.................................................................
-!     Calculate the integrals prior to using them
+!     Calculate the integrasetup0%ls prior to using them
```
again

garrettwrong commented on cfpcoefr.f90:
```diff
@@ -449,13 +449,13 @@ subroutine cfpcoefr
 !     The do loops 302 through 307 seek to take advantage of the
 !     Maxwellian nature of f for v < vth/2. This is employed in
-!     the integrations to obtain the functionals - Thus f is
+!     the integrations to obtain the functionasetup0%ls - Thus f is
```
again

garrettwrong commented on cfpcoefr.f90:
```diff
@@ -523,7 +523,7 @@ subroutine cfpcoefr
                   jp=jj+1
                   jm=j-1
-                  ! X-integrals after Eq.(53) Franz Thesis
+                  ! X-integrasetup0%ls after Eq.(53) Franz Thesis
```
more

garrettwrong commented on cfpcoefr.f90:
```diff
@@ -533,7 +533,7 @@ subroutine cfpcoefr
 !..................................................................
 !     In this set of loops I need to sum up all the contributions to the
-!     integrals and then just use these in the coefficients
+!     integrasetup0%ls and then just use these in the coefficients
```
more

garrettwrong commented on coefwti.f90:
```diff
@@ -152,7 +152,7 @@ subroutine coefwti(k)
 
 !.......................................................................
-!     Ensures correct differentiation at end of intervals
+!     Ensures correct differentiation at end of intervasetup0%ls
```
junk

garrettwrong commented on coefwtj.f90:
```diff
@@ -117,7 +117,7 @@ subroutine coefwtj(k)
 
 !.......................................................................
-!     Ensures correct differentiation at end of intervals
+!     Ensures correct differentiation at end of intervasetup0%ls
```
junk

garrettwrong commented on cqlconf.f90:
```diff
@@ -0,0 +1,284 @@
+    ! default logic sprinkled around the code, and condense
+    ! it here
+
+    if (present(mnemonic)) then
```
If we can get all the defaults moved to this module, we ditch the default branches in favor of a default instance of the type, which will be able to exist before the nml read, and also here.  Much nicer way to do it...

garrettwrong replied:
done, much cleaner.

garrettwrong commented on diag.f90:
```diff
@@ -18,7 +18,7 @@ subroutine diag
 
 !..................................................................
-!     This routine controls the calculation of various
+!     This routine controsetup0%ls the calculation of various
```
junk

garrettwrong commented on diaggnde.f90:
```diff
@@ -59,13 +60,13 @@ subroutine diaggnde
 
       if (oldiag .ne. "enabled") then
-!     use Legendre decomposition for computing the integrals
+!     use Legendre decomposition for computing the integrasetup0%ls
```
junk

garrettwrong commented on diaggnde.f90:
```diff
@@ -59,13 +60,13 @@ subroutine diaggnde
 
 !..................................................................
-!     Compute integrals for various moments
+!     Compute integrasetup0%ls for various moments
```
again

garrettwrong commented on diaggnde.f90:
```diff
@@ -76,13 +77,13 @@ subroutine diaggnde
-!     For cqlpmod.ne."enabled", default lmdpln_ will take on values
-!       lrz:1 and successive calls (for given time step).  l_ also
+!     For setup0%cqlpmod.ne."enabled", default lmdpln_ will take on values
+!       setup0%lrz:1 and successive calsetup0%ls (for given time step).  l_ also
```
here

garrettwrong commented on diaggnde.f90:
```diff
@@ -105,7 +106,7 @@ subroutine diaggnde
                ! can become unstable in such case.
                ! So, keep f() non-zero in the loss cone,
-               ! but do not add such f() to the integrals/sums over d3v.
+               ! but do not add such f() to the integrasetup0%ls/sums over d3v.
```
here

garrettwrong commented on diaggnde.f90:
```diff
@@ -259,7 +260,7 @@ subroutine diaggnde
           gni=1./gn
           hni=0.0
-          !XXXXXX infinity?  YuP: hn and gn are never 0 - they are integrals over distr.func.
+          !XXXXXX infinity?  YuP: hn and gn are never 0 - they are integrasetup0%ls over distr.func.
```
here

garrettwrong commented on diaggnde2.f90:
```diff
@@ -42,7 +43,7 @@ subroutine diaggnde2
 
 !..................................................................
-!l    1. Compute integrals for 0th and 1st moments using the same
+!l    1. Compute integrasetup0%ls for 0th and 1st moments using the same
```
junk

garrettwrong commented on eqcoord.f90:
```diff
@@ -24,7 +24,7 @@ subroutine eqcoord
 
 !..................................................................
-!     This routine controls the translation of the equilibrium psi
+!     This routine controsetup0%ls the translation of the equilibrium psi
```
junk

garrettwrong commented on eqrhopsi.f90:
```diff
@@ -74,13 +74,13 @@ subroutine eqrhopsi(generate)
     !
-    !     Calls to the current subroutine eqrhospi are from eqcoord, in
-    !     descending order lr_=lrzmax,1,-1.
+    !     Calsetup0%ls to the current subroutine eqrhospi are from eqcoord, in
```
junk

garrettwrong commented on eqtopeol.f90:
```diff
@@ -109,7 +109,7 @@ subroutine eqtopeol
 
 !..................................................................
-!     Integrate with mstep subintervals
+!     Integrate with mstep subintervasetup0%ls
```
junk

garrettwrong commented on eqwrng.f90:
```diff
@@ -49,7 +49,7 @@ subroutine eqwrng(kerr)
 
 10010 format("subroutine eqorbit - do loop 10 error")
-10020 format("subroutine eqorbit - labels 15 to 20 - no convergence")
+10020 format("subroutine eqorbit - labesetup0%ls 15 to 20 - no convergence")
```
junk

garrettwrong commented on exsweep.f90:
```diff
@@ -85,7 +85,7 @@ subroutine exsweep
 !     except at the pass/trapped boundary) certain coefficients
 !     are zeroed out or suitably averaged at specific mesh points.
-!     The numbers 1,2,3 appearing in the calls below signify
+!     The numbers 1,2,3 appearing in the calsetup0%ls below signify
```
junk

garrettwrong commented on finit.f90:
```diff
@@ -290,7 +291,7 @@ subroutine finit
 !     delete the particles on loss orbits..
       if (lossmode(k) .eq. "mirrorcc") then
-         ! YuP: how about other models, "mirrsnk" and "mirrsnk1" ?
+         ! YuP: how about other modesetup0%ls, "mirrsnk" and "mirrsnk1" ?
```
oops

garrettwrong commented on fle.f90:
```diff
@@ -34,7 +34,7 @@ subroutine fle_pol(setup,lp)
 !  Presently set up only for single flux surface runs.
 !
-!  Normalization of the output distribution fl is such that it equals
+!  Normalization of the output distribution fl is such that it equasetup0%ls
```
oops

garrettwrong commented on fle.f90:
```diff
@@ -237,15 +237,15 @@ subroutine fle_fsa(setup)
 !  This calculation is set up for use with multiple flux surface runs.
 !
-!  Normalization of the output distribution fl is such that it equals
+!  Normalization of the output distribution fl is such that it equasetup0%ls
```
oops

garrettwrong commented on fle.f90:
```diff
@@ -400,7 +400,7 @@ subroutine fle(setup,lp)
 !    at normalized velocity xlm(1:jfl-1).
 !
-!  Normalization of the output distribution fl is such that it equals
+!  Normalization of the output distribution fl is such that it equasetup0%ls
```
oops

garrettwrong commented on ilut.f90:
```diff
@@ -81,7 +81,7 @@ subroutine ilut(n,a,ja,ia,lfil,droptol,alu,jlu,ju,iwk,w,jw,ierr)
 !
 ! droptol = real(c_double). Sets the threshold for dropping small terms in the
-!           factorization. See below for details on dropping strategy.
+!           factorization. See below for detaisetup0%ls on dropping strategy.
```
oops

garrettwrong commented on ilut.f90:
```diff
@@ -141,7 +141,7 @@ subroutine ilut(n,a,ja,ia,lfil,droptol,alu,jlu,ju,iwk,w,jw,ierr)
 ! (however, fill-in is then mpredictible).                             *
 !----------------------------------------------------------------------*
-!     locals
+!     locasetup0%ls
```
oops

garrettwrong commented on ilut.f90:
```diff
@@ -422,7 +422,7 @@ subroutine pgmres(n, im, rhs, sol, vv, eps, maxits, iout, &
 ! This is a simple version of the ILUT preconditioned GMRES algorithm. *
 ! The ILUT preconditioner uses a dual strategy for dropping elements   *
-! instead  of the usual level of-fill-in approach. See details in ILUT *
+! instead  of the usual level of-fill-in approach. See detaisetup0%ls in ILUT *
```
oops

garrettwrong commented on impchk.f90:
```diff
@@ -24,7 +24,7 @@ subroutine impchk(k)
 !     This routine computes the density gain (from dF/dt) and compares i
 !     locally to the density gain from the r.h.s. of the F.P. equation.
-!     sumleft and sumright are the integrated totals. error is the
+!     sumleft and sumright are the integrated totasetup0%ls. error is the
```
oops

garrettwrong commented on lookup.f90:
```diff
@@ -24,7 +24,7 @@ subroutine lookup(x,xarray,length,weightu,weightl,lement)
 !     weightu/weightl are the upper/lower weights for xarray
 !     in  x(lement:lement+1).
-!     If x falls outside bounds of the array, weightl/weightu set
+!     If x falsetup0%ls outside bounds of the array, weightl/weightu set
```
oops

garrettwrong commented on lookup.f90:
```diff
@@ -91,7 +91,7 @@ subroutine lookup_tdf(x,xarray,length,weightu,weightl,lement)
 !     weightu/weightl are the upper/lower weights for xarray
 !     in  x(lement:lement+1).
-!     If x falls outside bounds of the array, weightl/weightu set
+!     If x falsetup0%ls outside bounds of the array, weightl/weightu set
```
oops

garrettwrong commented on losscone.f90:
```diff
@@ -28,7 +28,7 @@ subroutine losscone
 !     [Subroutine called from ainitial. indxlr_ is set by tdnflxs(ll)
 !      in a loop ll=lrors,1,-1 in tdinitl.]
-!     Several loss models are available and are invoked by setting
+!     Several loss modesetup0%ls are available and are invoked by setting
```
oops

garrettwrong commented on losscone.f90:
```diff
@@ -220,7 +220,7 @@ subroutine losscone
 !                     endif  ! on lossmode(k).eq.'simplbn2'
 
-!BH160529 There are problems with application of NB for these models.
+!BH160529 There are problems with application of NB for these modesetup0%ls.
```
oops

garrettwrong commented on losstor.f90:
```diff
@@ -17,7 +17,7 @@ subroutine losstor(k)
 
 !..................................................................
-!     This routine provides a number of phenomenological loss models
+!     This routine provides a number of phenomenological loss modesetup0%ls
```
oops

garrettwrong commented on micgnbnd.f90:
```diff
@@ -26,7 +26,7 @@ subroutine micgnbnd
 !     on the p/t boundary (y(itl,l_)) but the singularity is
 !     integrable.
-!     See the user manual for mathematical details.
+!     See the user manual for mathematical detaisetup0%ls.
```
oops

garrettwrong commented on micxinil.f90:
```diff
@@ -305,10 +305,10 @@ subroutine micxinil
 
 !..................................................................
-!     compute legendre polynomials and their derivatives along
+!     compute legendre polynomiasetup0%ls and their derivatives along
```
oops

garrettwrong commented on micxinit.f90:
```diff
@@ -322,15 +322,15 @@ subroutine micxinit
 !.......................................................................
 !l    3.1 Define the mesh point centered integration coefficients
-!     employed for orbit field line integrals, z mesh defined in micxiniz
+!     employed for orbit field line integrasetup0%ls, z mesh defined in micxiniz
```
oops

garrettwrong commented on netcdfrw2.f:
```diff
@@ -140,6 +140,13 @@ subroutine netcdfrw2(kopt)
       data start3/1,1,1/
       data start4/1,1,1,1/
 
+
```
since I cheated here, I should remove all `setup0%` below

garrettwrong commented on pltdf.f90:
```diff
@@ -335,7 +337,7 @@ subroutine pltcont(k,pltcase,tt_,itype)
 !
 !     In the case abs(dmin).le.contrmin*dmax, then plot contours
-!     occur at levels:
+!     occur at levesetup0%ls:
```
oops

garrettwrong commented on pltdf.f90:
```diff
@@ -469,7 +471,7 @@ subroutine pltcont(k,pltcase,tt_,itype)
       !write(*,*)'FLOGMIN,FLOGMAX=',FLOGMIN,FLOGMAX
       !dmin, dmax are found above, for TEMP1(I,J)=f(I,J) (lin. scale).
-      !      but RCONT() levels are log()
+      !      but RCONT() levesetup0%ls are log()
```
oops

garrettwrong commented on pltlosc.f90:
```diff
@@ -13,18 +13,19 @@ module pltlosc_mod
 !     Plot contours of the loss region..
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -45,16 +45,17 @@ module pltmain_mod
     save
     !
-    !     This routine controls driver plots
+    !     This routine controsetup0%ls driver plots
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -45,16 +45,17 @@ module pltmain_mod
     !
     !
-    !     Modified some Graflib to pgplot calls by Yuri Petrov, 090727,
+    !     Modified some Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -479,7 +480,7 @@ subroutine  gptx2d(text)
     !                   with the horizontal, increasing counter-clockwise
     !                   (0.0 is horizontal).
-    ! FJUST  (input)  : controls horizontal justification of the string.
+    ! FJUST  (input)  : controsetup0%ls horizontal justification of the string.
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -500,14 +502,14 @@ subroutine pltends
 !     A cqlp.eq.'enabled' routine.
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -737,7 +739,7 @@ subroutine pltfluxs
 !...................................................................
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -834,7 +836,7 @@ subroutine pltfluxs
 !     except at the pass/trapped boundary) certain coefficients
 !     are zeroed out or suitably averaged at specific mesh points.
-!     The numbers 1,2,3 appearing in the calls below signify
+!     The numbers 1,2,3 appearing in the calsetup0%ls below signify
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -1087,7 +1090,7 @@ subroutine pltprppr
 !...
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -1223,14 +1227,14 @@ subroutine pltrstv
 !
 
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -1406,7 +1411,7 @@ subroutine pltstrml
 !     included and when the problem is at steady state.
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -1530,7 +1535,7 @@ subroutine pltstrml
 !     except at the pass/trapped boundary) certain coefficients
 !     are zeroed out or suitably averaged at specific mesh points.
-!     The numbers 1,2,3 appearing in the calls below signify
+!     The numbers 1,2,3 appearing in the calsetup0%ls below signify
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -1729,7 +1734,7 @@ subroutine pltstrml
             DMIN=MIN(temp1(I,J),DMIN)
             DMAX=MAX(temp1(I,J),DMAX)
-            RTEMP1(I,J)=temp4(I,J) ! the cont() levels of this function
+            RTEMP1(I,J)=temp4(I,J) ! the cont() levesetup0%ls of this function
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -1876,7 +1881,7 @@ subroutine pltfofvv
 !
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on pltmain.f90:
```diff
@@ -2031,9 +2036,9 @@ subroutine souplt
 
 !BH171231         write(t_,540) crnt_nbi
-!BH171231 540     format("NBI source rate=",1pe11.4," ptcls/cc/sec")
+!BH171231 540     format("NBI source rate=",1pe11.4," ptcsetup0%ls/cc/sec")
```
oops several same

garrettwrong commented on profiles.f90:
```diff
@@ -527,15 +527,15 @@ subroutine profiles
 
 !--------------------------------------------------------------------
-      if (ipronn.eq."spline-t") then ! time-dep. neutrals or impurities
+      if (ipronn.eq."spline-t") then ! time-dep. neutrasetup0%ls or impurities
```
oops

garrettwrong commented on prpprctr.f90:
```diff
@@ -28,7 +28,7 @@ subroutine prpprctr
 !
 !
-!     Modified from Graflib to pgplot calls by Yuri Petrov, 090727,
+!     Modified from Graflib to pgplot calsetup0%ls by Yuri Petrov, 090727,
```
oops

garrettwrong commented on psif.f90:
```diff
@@ -25,7 +25,7 @@ real(c_double) function psif(xz)
     !mnt  calculates the ratio of mod B at arclength from the midplane
     !mnt  to mod B at the midplane; 1 < psi < psimx(lr_).
-    !mnt  the value of psimodel controls which model is expressed:
+    !mnt  the value of psimodel controsetup0%ls which model is expressed:
```
oops

garrettwrong commented on r8subs.f90:
```diff
@@ -619,10 +619,10 @@ SUBROUTINE DGBTF2( M, N, KL, KU, AB, LDAB, IPIV, INFO )
 !*
 !*  KL      (input) INTEGER
-!*          The number of subdiagonals within the band of A.  KL >= 0.
+!*          The number of subdiagonasetup0%ls within the band of A.  KL >= 0.
```
oops

garrettwrong commented on r8subs.f90:
```diff
@@ -631,11 +631,11 @@ SUBROUTINE DGBTF2( M, N, KL, KU, AB, LDAB, IPIV, INFO )
 !*
-!*          On exit, details of the factorization: U is stored as an
-!*          upper triangular band matrix with KL+KU superdiagonals in
+!*          On exit, detaisetup0%ls of the factorization: U is stored as an
```
oops

garrettwrong commented on sigv.f90:
```diff
@@ -23,7 +23,7 @@ subroutine sigv(icall)
 
 !..................................................................
-!     sets up calls to sigma-v calculator
+!     sets up calsetup0%ls to sigma-v calculator
```
oops

garrettwrong commented on sigv5d.f90:
```diff
@@ -158,7 +158,7 @@ subroutine sigv5d
         sum=0.
 
-!     Loop over Legendre polynomials
+!     Loop over Legendre polynomiasetup0%ls
```
oops

garrettwrong commented on soup.f90:
```diff
@@ -36,7 +36,7 @@ subroutine soup(cosi,l,kk,m)
 !..................................................................
 !     For the case that a Gaussian profile in polar coordinates
-!     is to be computed determine some exponentials depending on
+!     is to be computed determine some exponentiasetup0%ls depending on
```
oops

garrettwrong commented on sourcee.f90:
```diff
@@ -33,7 +33,7 @@ subroutine sourcee
 !     The "sou" routines are independent of the "fr" routines and vice-
 !     versa.
-!     Also, calls Knock On source modules, if specified.
+!     Also, calsetup0%ls Knock On source modules, if specified.
```
oops

garrettwrong commented on tdchief.f90:
```diff
@@ -92,11 +95,14 @@ subroutine tdchief
 !..................................................................
-!     This routine directs the calculation of CQL3d; it controls
-!     input, output, calls to CQL, calls to WD*, and holds the
+!     This routine directs the calculation of CQL3d; it controsetup0%ls
```
oops

garrettwrong commented on tdchief.f90:
```diff
@@ -139,50 +146,54 @@ subroutine tdchief
+!-BH:  Except that some setup0%lrzmax=1 test cases are still used, and
 !-BH:  errors can arise when only only one value of a mesh
-!-BH:  is called  for.  For example, tdrmshst fails.
+!-BH:  is called  for.  For example, tdrmshst faisetup0%ls.
```
oops

garrettwrong commented on tdchief.f90:
```diff
@@ -518,10 +529,10 @@ subroutine tdchief
                ! If kelec>1, we can get a problem here.
                write(*,'(a,3i4,3e12.4)') &
-                 'after achiefn(3): n,it,ll,integrals f, f-h, g:', &
+                 'after achiefn(3): n,it,ll,integrasetup0%ls f, f-h, g:', &
```
oops

garrettwrong commented on tdinlegw.f90:
```diff
@@ -73,13 +73,13 @@ subroutine tdinlegw(klpar,klrad,klindxr,klmesh)
       if (analegco .eq. "enabled") go to 200
 
-!     CQL3D: imax points in two intervals [1,imax] and [iy+1-imax,iy]
+!     CQL3D: imax points in two intervasetup0%ls [1,imax] and [iy+1-imax,iy]
```
oops

garrettwrong commented on tdinlegw.f90:
```diff
@@ -28,7 +28,7 @@ subroutine tdinlegw(klpar,klrad,klindxr,klmesh)
 !     The integral is computed using a ngauss-point Gaussian quadrature
 !     where f at the Gauss points is obtained using a nlagran-point
-!     Lagrangian interpolation with polynomials (see 25.2.2 p.878,
+!     Lagrangian interpolation with polynomiasetup0%ls (see 25.2.2 p.878,
```
oops

garrettwrong commented on tdnpacxcs.f90:
```diff
@@ -29,7 +29,7 @@ REAL FUNCTION cxcs(enin)
 
         en=enin*1000 !converts to eV
-! uses chebyshev polynomials
+! uses chebyshev polynomiasetup0%ls
```
oops

garrettwrong commented on tdnpacxcs.f90:
```diff
@@ -48,7 +48,7 @@ REAL FUNCTION cxcs(enin)
        x=(dum1-dum2)/dum3
 !       print *,x
-! The chebyshev polynomials
+! The chebyshev polynomiasetup0%ls
```
we dont disrespect chebyshev

garrettwrong commented on tdnpalam.f90:
```diff
@@ -42,7 +42,7 @@ subroutine lam(en_m,ene,stoplamda)
        stopcs=0.
        call bscs(en_m(mm)/bnumb(1),ene(nn), &
-        temp(kelec,nn),zeff(nn),stopcs) ! calls bscs to get cross section
+        temp(kelec,nn),zeff(nn),stopcs) ! calsetup0%ls bscs to get cross section
```
oops

garrettwrong commented on tdrmshst.f90:
```diff
@@ -391,13 +392,13 @@ subroutine tdrmshst
-      ! Reason: the cumulative integrals over dlpsii(r),
-      ! effectively the integrals over r*dr,
+      ! Reason: the cumulative integrasetup0%ls over dlpsii(r),
+      ! effectively the integrasetup0%ls over r*dr,
```
oops

garrettwrong commented on tdsetnpa.f90:
```diff
@@ -27,7 +27,7 @@ subroutine tdsetnpa(sigmanpa)
 !     It is much simpler than required for the SXR case, due to the
 !     delta-functions in energy and scattering angle in the differential
-!     cross-section, assuming zero-velocity background neutrals.
+!     cross-section, assuming zero-velocity background neutrasetup0%ls.
```
oops

garrettwrong commented on tdsetsxr.f90:
```diff
@@ -37,7 +37,7 @@ subroutine tdsetsxr(ien)
 !
 
-!     Compute cos and sin arrays. Determine Legendre polynomials out to
+!     Compute cos and sin arrays. Determine Legendre polynomiasetup0%ls out to
```
oops

garrettwrong commented on tdtravct.f90:
```diff
@@ -66,23 +67,23 @@ subroutine tdtravct(f1,kprofile,ktransp)
 
 !..............................................................
-!     Compute two integrals that are defined on mid-mesh points.
+!     Compute two integrasetup0%ls that are defined on mid-mesh points.
```
oops

garrettwrong commented on tdtrvtor2.f90:
```diff
@@ -27,15 +27,15 @@ subroutine tdtrvtor2(f1,f2,vp,vp_,kopt)
 !     That is: sum(f1(i)*vp(i)) = sum(f2(i)*vp_(i)), i=1,iy_(l)
 !
-!     This new version of tdtrvtor is such that two successive calls
+!     This new version of tdtrvtor is such that two successive calsetup0%ls
```
oops

garrettwrong commented on tdtrvtor3.f90:
```diff
@@ -27,15 +27,15 @@ subroutine tdtrvtor3(f1,f2,vp,vp_,kopt,k)
 !     That is: sum(f1(i)*vp(i)) = sum(f2(i)*vp_(i)), i=1,iy_(l)
 !
-!     This new version of tdtrvtor is such that two successive calls
+!     This new version of tdtrvtor is such that two successive calsetup0%ls
```
oops

garrettwrong commented on tdwrng.f90:
```diff
@@ -65,7 +65,7 @@ subroutine tdwrng(kerr)
 10080 format("transport model- meshy=fixed_mu.  Insufficient resolution" &
         "in the trapped region for some l. Set tfac larger")
-10090 format("transport model - transport mesh falls EXACTLTY on p/t",/, &
+10090 format("transport model - transport mesh falsetup0%ls EXACTLTY on p/t",/, &
```
oops

garrettwrong commented on tdxinitl.f90:
```diff
@@ -22,13 +22,14 @@ module tdxinitl_mod
       save
 !..................................................................
-!     This routine "fills in" input data parabolically. It also
+!     This routine "filsetup0%ls in" input data parabolically. It also
```
oops

garrettwrong commented on urfb0.f90:
```diff
@@ -63,7 +63,7 @@ subroutine urfb0
 !**bh050820:  accounts for bounce time being twice the transitting bounce
 !**bh091031:  time.  However, bounce-averages are divided by bounce times,
-!**bh091031:  so this trapfac factor cancels out.
+!**bh091031:  so this trapfac factor cancesetup0%ls out.
```
oops

garrettwrong commented on urfb0.f90:
```diff
@@ -534,7 +534,7 @@ subroutine urfb_add(anpar0,vpar_res, &
 !**bh050820:  accounts for bounce time being twice the transitting bounce
 !**bh091031:  time.  However, bounce-averages are divided by bounce times,
-!**bh091031:  so this trapfac factor cancels out.
+!**bh091031:  so this trapfac factor cancesetup0%ls out.
```
oops

garrettwrong commented on urfb0.f90:
```diff
@@ -598,7 +598,7 @@ subroutine urfb_add(anpar0,vpar_res, &
       prf_rayel=0.d0 !sum-up contribution from a given ray el., per unit delpwr
 
-      do j=jmn,jx ! Levels of u in res.region, local u-space
+      do j=jmn,jx ! Levesetup0%ls of u in res.region, local u-space
```
oops

garrettwrong commented on urfb0.f90:
```diff
@@ -900,7 +900,7 @@ subroutine urfb_add(anpar0,vpar_res, &
                  urfb_i(i,l_)= urfb_i(i,l_)+delbw
                  urfc_i(i,l_)= urfc_i(i,l_)+delcc
-                 !Accumulate contributions in (i,l_) midplane grid cells:
+                 !Accumulate contributions in (i,l_) midplane grid celsetup0%ls:
```
oops

garrettwrong commented on urfb0.f90:
```diff
@@ -933,7 +933,7 @@ subroutine urfb_add(anpar0,vpar_res, &
                  ! This is to be saved into urfpwr(), the fractional power
                  ! (per delpwr) (contribution from a given ray element only)
-                 !Accumulate contributions in (i,l_) midplane grid cells:
+                 !Accumulate contributions in (i,l_) midplane grid celsetup0%ls:
```
oops

garrettwrong commented on urfb0.f90:
```diff
@@ -1043,7 +1043,7 @@ integer function luf_bin(px,parray)
          k1=k2 ! new k1  !  k3=k3 remains same
          goto 5 !-> divide this interval, and check again
-      else  ! Neither of two intervals => no index found.
+      else  ! Neither of two intervasetup0%ls => no index found.
```
oops

garrettwrong commented on urfbes.f90:
```diff
@@ -26,7 +26,7 @@ subroutine urfbes
 !     Routine computes three Bessel tables, J_nharm-1,J_nharm, and
 !     J_nharm+1. Space is allocated. Once tables are created, subsequent
-!     calls check to see if the maximum argument has increased beyond
+!     calsetup0%ls check to see if the maximum argument has increased beyond
```
oops

garrettwrong commented on urfchief.f90:
```diff
@@ -23,19 +23,20 @@ module urfchief_mod
 
 !..................................................................
-!     This routine controls the urf module
+!     This routine controsetup0%ls the urf module
```
oops

garrettwrong commented on urfchief.f90:
```diff
@@ -23,19 +23,20 @@ module urfchief_mod
 !     nrfstep1, nrfstep2, nrfpwr, nrfitr1, nrfitr2, nrfitr3, urfncoef
 !
-!     nurf=a counter for the number of calls to urfchief which have
+!     nurf=a counter for the number of calsetup0%ls to urfchief which have
```
oops

garrettwrong commented on urfchief.f90:
```diff
@@ -50,14 +51,14 @@ subroutine urfchief
 !     = (1/2)**nrfpwr.
 !     return.
-!     3. Repeat step 2 for next nrfpwr calls, but with fractional input
+!     3. Repeat step 2 for next nrfpwr calsetup0%ls, but with fractional input
```
three more here

garrettwrong commented on urfdamp1.f90:
```diff
@@ -58,7 +58,7 @@ subroutine urfdamp1(krf)
 !**bh050820:  accounts for bounce time being twice the transitting bounce
 !**bh091031:  time.  However, bounce-averages are divided by bounce times,
-!**bh091031:  so this trapfac factor cancels out.
+!**bh091031:  so this trapfac factor cancesetup0%ls out.
```
oops

garrettwrong commented on urfrays.f90:
```diff
@@ -17,7 +17,7 @@ subroutine urfrays(initrf,nrayptlh,nrayptec,nrayptfw)
 
 !.................................................................
-!     This routine calls ray tracing code(s), if namelist
+!     This routine calsetup0%ls ray tracing code(s), if namelist
```
oops

garrettwrong commented on zcunix.f90:
```diff
@@ -128,13 +128,13 @@ end function taper
 !
 !     usage                  for one-dimensional cubic spline interpolation,
-!     the user first calls coeff1 by
+!     the user first calsetup0%ls coeff1 by
```
oops

garrettwrong commented on zcunix.f90:
```diff
@@ -1680,7 +1680,7 @@ subroutine zzrprt(nchars,narray)
       nch = 10
 !mnt  for line printers, use
-!mnt  for data terminals, use
+!mnt  for data terminasetup0%ls, use
```
oops

garrettwrong commented on zcunix.f90:
```diff
@@ -204,7 +204,7 @@ end function taper
 !     derivative and incorporates the given
 !     boundary conditions.  coeff1  sets up this
-!     system and calls subroutine trip to solve it.
+!     system and calsetup0%ls subroutine trip to solve it.
```
oops to here

garrettwrong commented on zcunix.f90:
```diff
@@ -1697,7 +1697,7 @@ subroutine zzstgt(k,nfatal,ntrace)
 
 !mnt  *** if local variables are not normally retained between
-!mnt  calls on this system, the variables lnf and lnt can be
+!mnt  calsetup0%ls on this system, the variables lnf and lnt can be
```
oops

garrettwrong commented on zcunix.f90:
```diff
@@ -1735,8 +1735,8 @@ subroutine zzxset(nfatal,ntrace)
 
-!mnt  *note* -- some calls to zzrchk will cause unconditional
-!mnt  termination of execution.  zzxset has no effect on such calls.
+!mnt  *note* -- some calsetup0%ls to zzrchk will cause unconditional
```
and finally these two. yay

ypetrov commented:
I am having trouble with compiling the new module cqlconf.f90.
cqlconf.f90(43) : Error: This symbol must be a defined parameter or an 
argument of an inquiry function that evaluates to a compile-time 
constant.   [SETUP0]
cqlconf.f90(43) : Error: This is not a valid initialization expression.  
  [SETUP0]
cqlconf.f90(42) : Error: This entity cannot be PUBLIC since its derived 
type is PRIVATE.   [SETUP0]

I guess my Intel compiler does not like this:
    type(setup0_t), pointer, private :: setup0_=>setup0 ! hack
Any idea how to fix this?

Yuri



On 2019-06-11 09:29, Garrett Wright wrote:
> @garrettwrong [1] requested your review on: #28 [2] implement setup0
> type in code base, updates makefile deps..
> 
> --
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub [3], or mute the
> thread [4].
> 
> Links:
> ------
> [1] https://github.com/garrettwrong
> [2] https://github.com/garrettwrong/cql3d_f90/pull/28
> [3]
> https://github.com/garrettwrong/cql3d_f90/pull/28?email_source=notifications&amp;email_token=ABO4SDXST52AAI5AVX7FEQTPZ7AGLA5CNFSM4HWXIUZKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOR5JPD4Q#event-2404577778
> [4]
> https://github.com/notifications/unsubscribe-auth/ABO4SDVQB5HE5QICKOADKWTPZ7AGLANCNFSM4HWXIUZA

garrettwrong commented:
Yes, there are plenty of ways to work around.

It actually doesn't need to happen, its just that I was trying to keep the NML group name and the type name the same in all the code, and existing nmls.

If it is not valid code, I will replace.  If it is valid code, we'll try figuring out why intel is unhappy, maybe version.

garrettwrong commented:
you can see the gnu build was successful, and reproduced results...

ypetrov commented:
Even if I rename 
  namelist/setup0/ …
to 
 namelist/setup00/ …
the error messages are still same.

garrettwrong commented:
yeah, that isn't the problem, you'd need to change other things yuri

I just read it is a fortran standard feature not yet supported by intel. Shame.  i'll replace it.

garrettwrong commented:
what version of intel do you use exactly? I can see about putting it into CI if they allow it.

ypetrov commented:
I use Intel Visual Fortran,  IVF-10.1.011

ypetrov commented:
The "Visual" part is not important. During compilation I see:
   Compiling with Intel(R) Fortran Compiler 10.1.011

garrettwrong commented:
yuri, for worrying about cpu cycles in predictable if statements, you seem to not care about running a recent compiler.  I'm not sure I can find one that old.  Do you use any others?

Thanks for pointing this out btw. Since I dropped the ainsetpa stuff and replaced it with defaults in the der type this morning, i actually didn't need to do that hack anymore at all... thanks.

You should still upgrade your compiler though, I think you would find newer ones have less trouble with anything post 95...

ypetrov commented:
Still getting one error message:
cqlconf.f90(43) : Error: This entity cannot be PUBLIC since its derived type is PRIVATE.   [SETUP0]

Updating Intel compiler will cost money...
Anyway, for long runs I usually use NERSC with 128-512 mpi cores.

ypetrov commented:
I guess I could use gfortran on PC.  But some users may have an old Intel Compiler, so it is good to test.

garrettwrong commented:
Yes fancy compilers are expensive.  I am not particularly impressed or interested in them myself.  Usually people that like them keep them up to date.  Intel is available free for personal use ;) (probably not the visual bit).

Took some time to check (our systems are _super slow_), but the code in question builds fine against intel 15-19 here at PPPL using a minimal adaptation of the centos makefile (NAME, COMPILER, DEBUG, LDSPECIAL).  Running it complained about pgplot (I doubt it was installed well, not either of our problems).

I went ahead and fixed what I think your compiler was complaining about anyway (below), for those who demand the finest in vintage compilers.

ypetrov commented:
Thank you! That compiled well. 
I still don't understand why some subroutines have 
 use cqlconf_mod, only : setup0
but some do not have it.
If setup0 is public, why do we need that line?
P.S. While you are on it, please replace 
"sols,solzz"  by "solrz,solzz"   - there are two entrances like this (in comments).

ypetrov commented:
in tdplteq.f90, 
this line:
      IF (LRZMAX.GT.200) STOP 'TDPLTEQ: CHECK DIM OF TEXTT'
Change LRZMAX to setup0%lrzmax

ypetrov commented:
There are many other upper case LRZMAX - missed from change to setup0%lrzmax

ypetrov commented:
Also 
pltendn.f90(581):            RLRZAP11(1:LRZ)= consnp(it,1:LRZ)
I guess many upper case vars from setup0 are missed.

garrettwrong commented:
perfect, would you like to try to submit a patch for those here?

ypetrov commented:
If you are busy, I can do it. But then it will require a review from you?

ypetrov commented:
Well, let me know who will do it. There are only four vars like this:
LRZ, LRZMAX, LS, LSMAX

garrettwrong commented:
You can do it, good practice (if you haven't already).

```
git fetch origin
git checkout nml_dertype
<make changes>
git commit -am "some message"
git push origin nml_dertype
```

if you mess it up I will just write over your changes and you can try again.

Also, if you are reviewing on the website, you can click a line and write a message there, and I will see what/where you mean.

ypetrov commented:
ok, I'll commit and push. Already did similar operations to my yuri_* branches.

garrettwrong commented on tdoutput.f90:
```diff
@@ -128,9 +128,9 @@ subroutine tdoutput(kopt)
-! xxx can we get past this line, then dies at 147
-!      WRITE(6,91111) 
-print * , l,rovera(l),eps(l),btor0(l),bthr0(l), &
+! xxx can we get past this line, then dies at 147 !YuP: not seeing any problem: i3, then 10 values e13.4
```
ahh, this was a debug comment I made to myself :) I'll remove

garrettwrong commented:
@ypetrov , when you (and/or Bob) are happy with these changes, you can `approve`.  I'll merge it in at that point.

You are doing well this time, nice job, thanks.

ypetrov commented:
All good.  I guess Bob has to approve now?

garrettwrong commented:
I talked to him today for a bit about this code.  He didn't have any immediate objections, so I think we can merge this and fix anything as minor updates.  Seems like you caught most of the places I missed in the first pass (_it was a lot of code to hit_).  Thanks, merging
